recipes/microservice-essentials/all/conanfile.py

```diff
@@ -0,0 +1,106 @@
+        "with_tests": [True, False],
+        "with_examples": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_tests": False,
+        "with_examples": False
+    }
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "Visual Studio": "15.7",
+        }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16.3 <4]")
+
+    def requirements(self):
+        if self.options.with_examples:
+            self.requires("cpp-httplib/0.12.4")
+            self.requires("nlohmann_json/3.11.2")
+            self.requires("openssl/3.1.0")
+            self.requires("grpc/1.50.1")
+        if self.options.with_tests:
+            self.requires("catch2/3.3.2")
+            self.requires("nlohmann_json/3.11.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        tc.variables["BUILD_TESTING"] = self.options.with_tests
+        tc.variables["BUILD_EXAMPLES"] = self.options.with_examples
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "17")
+
+        def loose_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and loose_lt_semver(str(self.settings.compiler.version), minimum_version):
+            raise ConanInvalidConfiguration(
+                "{} requires C++17, which your compiler does not support.".format(self.name)
+            )
+    
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["microservice-essentials"]
+        if self.settings.os != "Windows":
+            self.cpp_info.system_libs = ["pthread"]
+        lib_path = os.path.join(self.package_folder, "lib")
+        self.output.info("Appending PATH environment variable with : {0}".format(lib_path))
+        self.runenv_info.prepend_path("PATH", lib_path)
```

Comment:
Why is this needed?

Reply:
Windows/Shared lib case: .dll is in the lib_path. The test_package (and any other user of the library) needs to be able to find the .dll file. Is there a better way to ensure this?

Reply:
Ok, I see. This is not needed because Conan will add those paths to the `VirtualRunEnv` generator and create a `conanrun.bat/sh` file so you can activate the environment before running the program built in shared mode. Please check this section of the tutorial in the Conan docs: https://docs.conan.io/2/tutorial/consuming_packages/different_configurations.html#modifying-options-linking-the-application-dependencies-as-shared-libraries

Reply:
unfortunately, just removing isn't enough. The dll needs to be in the bin folder for this to work. I now explicitly copied the files into this folder.